osdep/timer-linux.c

```diff
@@ -34,22 +32,16 @@ void mp_sleep_us(int64_t us)
     nanosleep(&ts, NULL);
 }
 
-#if defined(_POSIX_TIMERS) && _POSIX_TIMERS > 0 && defined(CLOCK_MONOTONIC)
-uint64_t mp_raw_time_us(void)
+uint64_t mp_raw_time_ns(void)
 {
-    struct timespec ts;
-    if (clock_gettime(CLOCK_MONOTONIC, &ts))
-        abort();
-    return ts.tv_sec * 1000000LL + ts.tv_nsec / 1000;
-}
+    struct timespec tp = {0};
+#if defined(CLOCK_MONOTONIC_RAW)
+    clock_gettime(CLOCK_MONOTONIC_RAW, &tp);
 #else
-uint64_t mp_raw_time_us(void)
-{
-    struct timeval tv;
-    gettimeofday(&tv,NULL);
```

Comment:
"Using gettimeofday() as a precise timepoint is not that good idea."
but as you can see here we're not doing that, it's merely a fallback

Reply:
Fair, will reword this commit message. We are still truncating,